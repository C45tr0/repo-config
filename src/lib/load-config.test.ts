import { chdirSetup } from "../../test/helpers/chdir";

import loadConfig from "./load-config";

const chdir = chdirSetup("load-config");

test("finds config in package.json", function findBasicConfig() {
  chdir("package-only");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [{ loc: "local", cmd: "random-command" }]
      },
      files: { "random-file": "random-file.txt" }
    }
  ]);
});

test("finds config at repo.config.json", function findBasicConfig() {
  chdir("config-only");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [{ loc: "local", cmd: "random-command" }]
      },
      files: { "random-file": "random-file.txt" }
    }
  ]);
});

test("finds custom config from package.json", function findBasicConfig() {
  chdir("custom-config");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [{ loc: "local", cmd: "random-command" }]
      },
      files: { "random-file": "random-file.txt" }
    }
  ]);
});

test("finds config in package.json and repo.config.json and merges", function findBasicConfig() {
  chdir("merge-config");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [{ cmd: "random-command", loc: "local" }],
        "random-command2": [
          { cmd: "random-command2", loc: "local" },
          { cmd: "random-command3", loc: "node_modules/test-extend" }
        ],
        "random-command3": [
          { cmd: "random-command3", loc: "node_modules/test-extend2" }
        ]
      },
      files: {
        "random-file": "random-file.txt",
        "random-file2": "random-file2.txt",
        "random-file3": "node_modules/test-extend2/random-file3.txt"
      }
    }
  ]);
});

test("recurses extends and merges", function findBasicConfig() {
  chdir("loads-extends");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [
          { cmd: "random-command", loc: "node_modules/test-extend" }
        ]
      },
      files: {
        "random-file": "node_modules/test-extend/random-file.txt",
        "random-file2": "node_modules/test-extend2/random-file2.txt"
      }
    }
  ]);
});

test("recurses extends and merges with hierarchy node_modules", function findBasicConfig() {
  chdir("loads-extends-hierarchy");

  expect(loadConfig()).toEqual([
    "./",
    {
      cmds: {
        "random-command": [
          { cmd: "random-command", loc: "node_modules/test-extend" }
        ]
      },
      files: {
        "random-file": "node_modules/test-extend/random-file.txt",
        "random-file2":
          "node_modules/test-extend/node_modules/test-extend2/random-file2.txt"
      }
    }
  ]);
});

test("invalid extension name", function findBasicConfig() {
  chdir("invalid-extension");

  expect(loadConfig).toThrowErrorMatchingSnapshot();
});
