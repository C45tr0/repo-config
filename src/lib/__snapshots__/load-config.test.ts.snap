// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid extension name 1`] = `"Extension not found: 'invalid-extension' paths searched: [\\"node_modules/invalid-extension\\",\\"node_modules/invalid-extension\\"]"`;
